### PR TITLE
Crude skeleton for spikeextractors producer generators

### PR DESCRIPTION
**Pull request by yarikoptic**
@jwodder  here is a crude idea for what I had in mind in #13 (drafting a PR as dirty as it was before I had to switch to another call/activity...) .  WDYT - could we implement it in some way like this?

### CODE REVIEW

**Comment by jwodder** [comment]
@yarikoptic I've changed the PR to check out the ephy_testing_data in a cached directory and also updated the rest of the code so that `create()` returns an iterable of triples.

File: nwb_healthstatus/producers/spikeextractors/recording.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    def create(self):
+        for se_class, dataset_path, se_kwargs in self.SCENARIOS:
+            self.dataset.get(dataset_path)
+            for k in ["filename", "file_or_folder_path", "file_path", "dirname", "folder_path"]:
+                if k in se_kwargs:
+                    se_kwargs[k] = self.pt / se_kwargs[k]
+            if "filename" in se_kwargs:
+                se_kwargs["filename"] = str(se_kwargs["filename"])
+            recording = se_class(**se_kwargs)
+            with tempfile.TemporaryFile() as tf:
```
Comment (yarikoptic) [inline nwb_healthstatus/producers/spikeextractors/recording.py]: this piece needs to be finished up

**Comment by yarikoptic** [comment]
Thank you @jwodder -- please finish it up to your liking (see the comment pointing to code to be finished) to get it all working and merge it whenever you are satisfied.

File: nwb_healthstatus/producers/spikeextractors/recording.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        for se_class, dataset_path, se_kwargs in self.SCENARIOS:
+            self.dataset.get(dataset_path)
+            for k in ["filename", "file_path", "dirname", "folder_path", "settings_file"]:
+                if k in se_kwargs:
+                    se_kwargs[k] = self.pt / se_kwargs[k]
+            if "filename" in se_kwargs:
+                se_kwargs["filename"] = str(se_kwargs["filename"])
+            recording = se_class(**se_kwargs)
+            nwbfile = NWBFile()
+            se.NwbRecordingExtractor.write_recording(recording, nwbfile=nwbfile, write_scaled=True)
```
Comment (jwodder) [inline nwb_healthstatus/producers/spikeextractors/recording.py]: @yarikoptic This bit doesn't seem to work; `nwbfile = NWBFile()` seems to be erroring out with `TypeError: NWBFile.__init__: missing argument 'session_description', missing argument 'identifier', missing argument 'session_start_time'`.

Comment (yarikoptic) [reply nwb_healthstatus/producers/spikeextractors/recording.py]: just add some dummy ones.  This is just for a test after all

File: nwb_healthstatus/producers/spikeextractors/recording.py [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+            sf = sorting.get_sampling_frequency()
+            if (
+                sf is None
+            ):  # need to set dummy sampling frequency since no associated acquisition in file
+                sf = 30000
+                sorting.set_sampling_frequency(sf)
+            with tempfile.NamedTemporaryFile() as tf:
+                se.NwbSortingExtractor.write_sorting(sorting, tf.name)
+                with pynwb.NWBHDF5IO(tf.name, mode="r") as io:
+                    nwbfile = io.read()
```
Comment (jwodder) [inline nwb_healthstatus/producers/spikeextractors/recording.py:280]: @yarikoptic I got the Extractors to run, but the Sorters are failing here with `hdmf.backends.io.UnsupportedOperation: Cannot read data from file /var/folders/l7/wrkq93d133d8zpn36fmqrq0r0000gn/T/tmpz8otn54x in mode 'r+'. There are no values.`.

Comment (yarikoptic) [reply nwb_healthstatus/producers/spikeextractors/recording.py:280]: is it only for a single extractor or for all of them?

Comment (yarikoptic) [reply nwb_healthstatus/producers/spikeextractors/recording.py:280]: if single -- just comment it out for now and produce for those for which it produces, and report for those which do not.
If for all -- I guess it is something more fundamental - just add a comment and do not enable the sorters tests

Comment (jwodder) [reply nwb_healthstatus/producers/spikeextractors/recording.py:280]: @yarikoptic It's for all of them.

**Pull request merged**